eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -80,8 +85,8 @@ private SubmittedVariantEntity getSubmittedVariantEntity(Document notClusteredSu
     public void open(ExecutionContext executionContext) throws ItemStreamException {
         MongoDatabase db = mongoClient.getDatabase(database);
         MongoCollection<Document> collection = db.getCollection(SUBMITTED_VARIANT_ENTITY);
-
-        Bson query = Filters.and(Filters.in(ASSEMBLY_FIELD, assembly));
+        Bson query = Filters.and(Filters.in(ASSEMBLY_FIELD, assembly),
+                Filters.exists(CLUSTERED_VARIANT_ACCESSION_FIELD, readOnlyClusteredVariants));
```

Comment:
I know this was not specified in the scope but I think we should also filter on the existence of the  `remappedFrom` field because we only want to cluster the remapped variant that have an RSid not the one that were created in this assembly.

Reply:
we have the filtering on **"remappedFrom"** field in java. 
If we want to put the filter at the DB level, we will have to separate out the query because this query is same for both mongo readers (clustered and non clustered reader). If we put it here, even in case of non-clustered it will try to search for that field.

Reply:
I see.